lib/TH/cmake/FindBLAS.cmake

```diff
@@ -242,32 +242,44 @@ endif()
 # Determine if blas was compiled with the f2c conventions
 IF (BLAS_LIBRARIES)
   SET(CMAKE_REQUIRED_LIBRARIES ${BLAS_LIBRARIES})
+  
   CHECK_C_SOURCE_RUNS("
 #include <stdlib.h>
 #include <stdio.h>
 float x[4] = { 1, 2, 3, 4 };
 float y[4] = { .1, .01, .001, .0001 };
-int four = 4;
-int one = 1;
+#ifdef WIN32
+  typedef __int64 BLINT;
+#else
+  typedef long BLINT;
```

Comment:
Are you sure this typedef is correct? It used to give `sdot_` an `int*` on non WIN32 platforms, now it uses `long*`, but `sizeof(long) != sizeof(int)` on most x86_64 systems. The same problem seems to be present in other parts of this PR too.